### ISSUE TITLE
force the rest of win32-select tests to pass

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#11152]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)| @glyph reported|
|-|-|
|Trac ID|trac#1645|
|Type|defect|
|Created|2006-04-16 04:49:36Z|

i.e. http://twistedmatrix.com/pipermail/twisted-python/2006-April/012858.html

I also implemented shebang-line support to make a new test, not mentioned in that message, pass: this requirement was introduced at r16614, with web2's CGI tests.

<details><summary>Searchable metadata</summary>

```
trac-id__1645 1645
type__defect defect
reporter__glyph glyph
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__win32__spawnProcess win32, spawnProcess
time__1145162976000000 1145162976000000
changetime__1145172947000000 1145172947000000
version__ 
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)|@glyph set owner to @jerub|
|-|-|

This branch is ready for review, in ```pyrrhic-1645```, as it is a pyrrhic victory over the Windows tests (the tests are unfixably buggy, so we won't have the satisfaction of actually fixing __behavior__ to make the code work properly...

**Comment by twisted-trac** [comment]
|[<img alt="jerub's avatar" src="https://avatars.githubusercontent.com/u/315461?s=50" width="50" height="50">](https://github.com/jerub)|@jerub set owner to @glyph|
|-|-|

review complete, please commit.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)|@glyph set status to `closed`|
|-|-|

(In [16628]) Fix win32 tests so the buildbot will turn green.

Fixes [#11152](https://github.com/twisted/twisted/issues/11152)

Author: glyph

Reviewer: jerub

As per discussion on the Twisted mailing list, turn off a few faililng tests to
make the tests pass on Windows.  Since that discussion, dreid added CGI tests
for ```web2``` which depend on shebang-line processing, so rather than wimp out
on those tests too, I added some simple shebang-line detection to the actual
process code and didn't change the tests at all.

This change also includes several fixes too small to bother with separate
commits.  It detects if the ```xattr``` module installed is the right one early
so tests won't fail if you have the "other" ```xattr``` installed.  Jerub also
noticed that the command-line quoting was pretty gross, so I refactored
```cmdLineQuote``` into ```quoteArguments```, since __every__ usage of
cmdLineQuote was simply calling ```' '.join(...)```.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
